src/emu/machines/MachineController.ts

```diff
@@ -249,6 +249,8 @@ export class MachineController implements IMachineController {
     // --- Start the machine and wait while it reaches the execution point
     const execInfo = this.machine.getMainExecPoint(codeToInject.model);
     await this.sendOutput("Initialize the machine", "blue");
+
+    this.isDebugging = debug;
```

Comment:
Without this the 'Is Debugging' store's got to receive an outdated value during the run method's update.